internal/configs/resource.go

```diff
@@ -647,6 +577,163 @@ func decodeReplaceTriggeredBy(expr hcl.Expression) ([]hcl.Expression, hcl.Diagno
 	return exprs, diags
 }
 
+// nolint: funlen, gocognit, cyclop, nolintlint // Legacy code moved from decode
```

Comment:
I understand it is not a brand-new function. However, would you mind refactoring it a bit to pass linting?